jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessor.java

```diff
@@ -145,6 +149,42 @@ public static JarType determineJarType(Path jarPath) throws IOException {
     return layers;
   }
 
+  /**
+   * Compute the entrypoint for a standard jar in exploded mode.
+   *
+   * @param jarPath path to jar file
+   * @param tempDirPath path to temporary jib local directory
+   * @param layers list of {@link FileEntriesLayer}
+   * @return list of {@link String} representing entrypoint
+   * @throws IOException if I/O error occurs when opening the jar file or if temporary directory
+   *     provided doesn't exist
+   * @throws IllegalArgumentException if main class is not found in the jar manifest
+   */
+  public static ImmutableList<String> computeEntrypoint_explodedStandard(
+      Path jarPath, Path tempDirPath, List<FileEntriesLayer> layers)
+      throws IOException, IllegalStateException {
+    Path localExplodedJarRoot = tempDirPath;
+    ZipUtil.unzip(jarPath, localExplodedJarRoot);
+
+    String mainClass = null;
+    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
+      mainClass = jarFile.getManifest().getMainAttributes().getValue(Attributes.Name.MAIN_CLASS);
+    }
+    if (mainClass == null) {
+      throw new IllegalArgumentException("Main-Class not found in jar's manifest.");
+    }
+    List<String> classpath = new ArrayList<>();
+    for (FileEntriesLayer layer : layers) {
+      if (layer.getName().equals(DEPENDENCIES) || layer.getName().equals(SNAPSHOT_DEPENDENCIES)) {
+        classpath.add(APP_ROOT.resolve("dependencies").toString());
```

Comment:
At least this should be
```suggestion
        classpath.add(APP_ROOT + "/dependencies/*");
```
Just so you know, there's an issue with the order of loading jars (like #1871 for Jib build plugins), but I think there's nothing we can do about it for a standard jar.

Reply:
is the ordering issue not a problem for a standard jar atm?